src/agent/border_agent.cpp

```diff
@@ -410,6 +424,28 @@ void BorderAgent::HandleExtPanId(void *aContext, int aEvent, va_list aArguments)
     static_cast<BorderAgent *>(aContext)->SetExtPanId(xpanid);
 }
 
+void BorderAgent::HandleThreadVersion(void *aContext, int aEvent, va_list aArguments)
+{
+    assert(aEvent == Ncp::kEventThreadVersion);
+
+    // `uint16_t` has been promoted to `int`.
+    uint16_t threadVersion = va_arg(aArguments, int);
+    static_cast<BorderAgent *>(aContext)->SetThreadVersion(threadVersion);
+}
+
+const char *BorderAgent::ThreadVersionToString(uint16_t aThreadVersion)
+{
+    switch (aThreadVersion)
+    {
+    case 2:
+        return "1.1.1";
+    case 3:
+        return "1.2.0";
```

Comment:
what if there is '1.2.1' someday? how to differentiate it from the same version value 3?

Reply:
see the [JIRA](https://threadgroup.atlassian.net/browse/BHC-541?atlOrigin=eyJpIjoiZTJlZjczMTVkNjgxNDgwYzhmMzY4ZjQyZjgyYjFlN2MiLCJwIjoiaiJ9) about this.

If the mDNS query response truely needs the `MAJOR.MINOR.PATCH` format of the Thread version, we need new API from OpenThread to provide this version string.

@jwhui

Reply:
I think it is OK to always return "1.2.0" if we can't decide the `PATCH`.